packages/eas-cli/src/devices/actions/create/currentMachineMethod.ts

```diff
@@ -0,0 +1,132 @@
+import spawnAsync from '@expo/spawn-async';
+import chalk from 'chalk';
+import os from 'os';
+
+import { ExpoGraphqlClient } from '../../../commandUtils/context/contextUtils/createGraphqlClient';
+import { AppleDeviceMutation } from '../../../credentials/ios/api/graphql/mutations/AppleDeviceMutation';
+import { AppleTeam } from '../../../graphql/generated';
+import Log from '../../../log';
+import { ora } from '../../../ora';
+import { confirmAsync } from '../../../prompts';
+import { DeviceCreateError } from '../../utils/errors';
+import {
+  DeviceData,
+  printDeviceData,
+  promptForDeviceClassAsync,
+  promptForNameAsync,
+} from './utils';
+
+export async function runCurrentMachineMethodAsync(
+  graphqlClient: ExpoGraphqlClient,
+  accountId: string,
+  appleTeam: Pick<AppleTeam, 'appleTeamIdentifier' | 'appleTeamName' | 'id'>
+): Promise<void> {
+  Log.newLine();
+  Log.log(chalk.white('Checking if current machine is an Apple Silicon one.'));
+  if (!isMachineAppleSilicon()) {
+    throw new DeviceCreateError(
+      "Current machine is not of Apple Silicon type - provisioning UDID can't be added automatically."
+    );
+  }
+  Log.log(chalk.green('Check successful.'));
+
+  await collectDataAndRegisterDeviceAsync(graphqlClient, { accountId, appleTeam });
+}
+
+function isMachineAppleSilicon(): boolean {
+  return os.cpus()[0].model.includes('Apple M') || os.arch() === 'arm64';
+}
+
+async function collectDataAndRegisterDeviceAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    accountId,
+    appleTeam,
+  }: {
+    accountId: string;
+    appleTeam: Pick<AppleTeam, 'appleTeamIdentifier' | 'appleTeamName' | 'id'>;
+  }
+): Promise<void> {
+  const deviceData = await collectDeviceDataAsync(appleTeam);
+  if (!deviceData) {
+    return;
+  }
+
+  const { udid, deviceClass, name } = deviceData;
+  const spinner = ora(`Registering Apple device on Expo`).start();
+  try {
+    await AppleDeviceMutation.createAppleDeviceAsync(
+      graphqlClient,
+      {
+        appleTeamId: appleTeam.id,
+        identifier: udid,
+        name,
+        deviceClass: deviceClass ?? undefined,
+      },
+      accountId
+    );
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+  spinner.succeed();
+}
+
+async function collectDeviceDataAsync(
+  appleTeam: Pick<AppleTeam, 'appleTeamIdentifier' | 'appleTeamName'>,
+  initialValues: Partial<DeviceData> = {}
+): Promise<DeviceData | null> {
+  Log.log(chalk.white('Fetching the provisioning UDID.'));
+  const [udid, defaultMachineName] = await fetchCurrentMachineUDIDAsync();
+  Log.log(chalk.green(`Fetched the provisioning UDID - ${udid}`));
+  const name = await promptForNameAsync(defaultMachineName ?? initialValues.name);
+  const deviceClass = await promptForDeviceClassAsync(initialValues.deviceClass);
```

Comment:
Should this be a case for the user to select a device class with us knowing it is MacBook? Maybe it should always be unknown? 🤔 

Is this device class something on Apple's side or something which exists on our backend? If it is the second case maybe we can add a MacBook class?

Reply:
I'm not sure about that, all of the existing methods use only IPad and IPhone. Adding a MacBook class makes sense to me but I'm not sure so far what the ramifications of that would be so I need to investigate some more.
From what I see `@expo/apple-utils` has more device classes so maybe it's ok

Reply:
In https://developer.apple.com/documentation/appstoreconnectapi/device/attributes there's also `MAC` option listed